src/evict/evict_lru.c

```diff
@@ -1227,9 +1227,10 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
     /*
      * If both queues are full and haven't been empty on recent refills, we're done.
      */
-    if (__evict_queue_full(queue) && cache->evict_empty_score < WT_EVICT_SCORE_CUTOFF)
+    if (__evict_queue_full(queue) && cache->evict_empty_score < WT_EVICT_SCORE_CUTOFF) {
```

Comment:
@y123456yz I'd prefer not to make this change. 
The `cache_eviction_queue_not_empty` [stat description](https://github.com/wiredtiger/wiredtiger/blob/7d0407c2904a4d1638478e459faa1157da4f8ef2/dist/stat_data.py#L331) is specifically for when we insert new pages into the queue ("when topping up") which this code path doesn't do. 
I'm also not sure what information we gain from incrementing the stat here. If we're hitting this branch then it can be derived from the fact that both `cache_eviction_queue_not_empty` and `cache_eviction_queue_empty` will be reported as zero.